src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let result = { ...state };
+  const resArray = [];
+
+  for (const act of actions) {
+    if (act.type === 'addProperties') {
+      Object.assign(result, act.extraData);
+      resArray.push({ ...result });
```

Comment:
remove all push expect the last one, let's push to result at the end of each iteration